src/Select.js

```diff
@@ -1,238 +1,116 @@
-        }
+    handleClear = e => {
+        e.stopPropagation()
+        this.props.onChange({})
     }
 
     render() {
         const {
-            dense,
-            disabled,
-            onChange,
-            value,
             children,
-            name,
             tabIndex,
-            valid,
-            warning,
-            error,
-            loading,
-            className,
+            selected,
+            placeholder,
+            clearable,
         } = this.props
-
-        const { focus } = this.state
+        const { open } = this.state
 
         return (
             <div
-                className={cx('select', className, {
-                    dense,
-                    disabled,
-                    focus,
-                    valid,
-                    warning,
-                    error,
-                })}
+                className="container"
+                ref={this.containerRef}
+                tabIndex={tabIndex}
+                onFocus={this.handleFocus}
+                onClick={this.handleOpen}
             >
-                <select
-                    onChange={onChange}
-                    value={value || -1}
-                    disabled={disabled}
-                    onFocus={this.onFocus}
-                    onBlur={this.onBlur}
-                    ref={this.selectRef}
-                    name={name}
-                    tabIndex={tabIndex}
-                >
-                    <option
-                        key="hidden-default-value"
-                        hidden
-                        disabled
-                        value="-1"
-                    />
-                    {children}
-                </select>
-
-                <div className="status-icon">
-                    <TailIcon />
-
-                    <StatusIconNoDefault
-                        error={error}
-                        valid={valid}
-                        loading={loading}
-                        warning={warning}
-                    />
-                </div>
-                <style jsx>{styles}</style>
+                <SelectInput
+                    placeholder={placeholder}
+                    selected={selected}
+                    clearable={clearable}
+                    onClear={this.handleClear}
+                    open={open}
+                />
+                {open && (
+                    <SelectDropdown
+                        onClick={this.handleOptionClick}
+                        selected={selected}
+                    >
+                        {children}
+                    </SelectDropdown>
+                )}
+
+                <style jsx>{`
+                    .container {
+                        position: relative;
+                        outline: 0;
+                    }
+                `}</style>
             </div>
         )
     }
 }
 
-Select.propTypes = {
-    className: propTypes.string,
+Select.defaultProps = {
```

Comment:
Not a biggie and nit picky, but normally defaultProps come after propTypes. But I don't care about this too much

Reply:
I'm used to the opposite actually, that's why I did this. I'll see what the other components do and follow that. Since it seems subjective I'll see what the existing code does and follow that.